identity/package.go

```diff
@@ -0,0 +1,33 @@
+/*
+Package identity provides code for parsing, storing and retrieving Red Hat Cloud identity
+from Go standard library context.
+
+To use Go HTTP middleware (handler), pass the EnforceIdentity function to the multiplexer:
+
+	r := mux.NewRouter()
+	r.Use(identity.EnforceIdentity)
+
+By default, both parsed and unparsed identities are stored in context.
+To extract identity or raw identity (base64 JSON string) from a context, use functions
+GetIdentity and GetRawIdentity:
+
+	id := identity.GetIdentity(ctx)
+	idRaw := identity.GetRawIdentity(ctx)
+
+The default middleware performs no logging. To plug the middleware into the application
+logging, use EnforceIdentityWithLogger:
+
+	func ErrorLogFunc(ctx context.Context, rawId, msg string) {
+		log := context.Value(myLoggerKey)
+		log.Errorf("Identity error: %s, raw identity: %s", msg, rawId)
+	}
+
+	// Go standard HTTP library example
+	handler := identity.EnforceIdentity(MyHandler())
+	handler.ServeHTTP(rr, req)
+
+	// Chi routing library example
+	r := mux.NewRouter()
+	r.Use(identity.EnforceIdentityWithLogger(ErrorLogFunc))
+*/
+package identity
```

Comment:
I don't like the idea of a separate "package.go" file for a package this small.
This package comment could easily exist at the top of `identity.go`. It's not
that large of a package. This keeps the documentation closer to the code too,
which is usually a good thing.

Reply:
Moved back.